osu.Game/Screens/Utility/LatencyArea.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input;
+using osu.Framework.Input.Events;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Utility
+{
+    public class LatencyArea : CompositeDrawable
+    {
+        [Resolved]
+        private OverlayColourProvider overlayColourProvider { get; set; } = null!;
+
+        public Action? ReportUserBest { get; set; }
+
+        private Drawable? background;
+
+        private readonly Key key;
+
+        public readonly int TargetFrameRate;
```

Comment:
Possibly would be more suited to use `int?` to indicate no limit rather than 0 as done currently.